soc/arm/cypress/psoc6/soc.c

```diff
@@ -0,0 +1,387 @@
+#endif
+#ifdef CY_CFG_SYSCLK_CLKPATH11_ENABLED
+	Cy_SysClk_ClkPath11Init();
+#endif
+#ifdef CY_CFG_SYSCLK_CLKPATH12_ENABLED
+	Cy_SysClk_ClkPath12Init();
+#endif
+#ifdef CY_CFG_SYSCLK_CLKPATH13_ENABLED
+	Cy_SysClk_ClkPath13Init();
+#endif
+#ifdef CY_CFG_SYSCLK_CLKPATH14_ENABLED
+	Cy_SysClk_ClkPath14Init();
+#endif
+#ifdef CY_CFG_SYSCLK_CLKPATH15_ENABLED
+	Cy_SysClk_ClkPath15Init();
+#endif
+
+	/* Configure and enable FLL */
+#ifdef CY_CFG_SYSCLK_FLL_ENABLED
+	Cy_SysClk_FllInit();
+#endif
+
+	/* Configure and enable PLLs */
+#ifdef CY_CFG_SYSCLK_PLL0_ENABLED
+	Cy_SysClk_Pll0Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL1_ENABLED
+	Cy_SysClk_Pll1Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL2_ENABLED
+	Cy_SysClk_Pll2Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL3_ENABLED
+	Cy_SysClk_Pll3Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL4_ENABLED
+	Cy_SysClk_Pll4Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL5_ENABLED
+	Cy_SysClk_Pll5Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL6_ENABLED
+	Cy_SysClk_Pll6Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL7_ENABLED
+	Cy_SysClk_Pll7Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL8_ENABLED
+	Cy_SysClk_Pll8Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL9_ENABLED
+	Cy_SysClk_Pll9Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL10_ENABLED
+	Cy_SysClk_Pll10Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL11_ENABLED
+	Cy_SysClk_Pll11Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL12_ENABLED
+	Cy_SysClk_Pll12Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL13_ENABLED
+	Cy_SysClk_Pll13Init();
+#endif
+#ifdef CY_CFG_SYSCLK_PLL14_ENABLED
+	Cy_SysClk_Pll14Init();
+#endif
+
+	/* Configure miscellaneous clocks */
+#ifdef CY_CFG_SYSCLK_CLKTIMER_ENABLED
+	Cy_SysClk_ClkTimerInit();
+#endif
+
+#ifdef CY_CFG_SYSCLK_CLKALTSYSTICK_ENABLED
+	Cy_SysClk_ClkAltSysTickInit();
+#endif
+
+#ifdef CY_CFG_SYSCLK_CLKPUMP_ENABLED
+	Cy_SysClk_ClkPumpInit();
+#endif
+
+#ifdef CY_CFG_SYSCLK_CLKBAK_ENABLED
+	Cy_SysClk_ClkBakInit();
+#endif
+
+	/* Configure default enabled clocks */
+#ifdef CY_CFG_SYSCLK_ILO_ENABLED
+	Cy_SysClk_IloInit();
+#else
+	Cy_SysClk_IloDisable();
+#endif
+
+#ifndef CY_CFG_SYSCLK_IMO_ENABLED
+#error the IMO must be enabled for proper chip operation
+#endif
+
+	/* Set accurate flash wait states */
+#if (defined(CY_CFG_PWR_ENABLED) && defined(CY_CFG_SYSCLK_CLKHF0_ENABLED))
+	Cy_SysLib_SetWaitStates(CY_CFG_PWR_USING_ULP != 0,
+		CY_CFG_SYSCLK_CLKHF0_FREQ_MHZ);
+#endif
+}
+/**
+ * Function Name: Cy_SystemInit
+ *
+ * \brief This function is called by the start-up code for the selected device.
+ * It performs all of the necessary device configuration based on the design
+ * settings. This includes settings for the platform resources and peripheral
+ * clock.
+ *
+ */
+
+void Cy_SystemInit(void)
+{
+	/* Configure platform resources */
+	init_cycfg_platform();
+
+	/* Configure peripheral clocks */
+	Cy_SysClk_PeriphSetDivider(CY_SYSCLK_DIV_8_BIT, 0u, 0u);
+	Cy_SysClk_PeriphEnableDivider(CY_SYSCLK_DIV_8_BIT, 0u);
+}
+
+/**
+ * Function Name: InitHardware()
+ *
+ *  Peforms hardware initialization: LEDs.
```

Comment:
Fix this comment block.  Function name doesnt match, reference to LEDs doesn't make sense.